ffai/ai/layers.py

```diff
@@ -313,9 +320,13 @@ def produce(self, game):
                         out[action_choice.positions[i].y][action_choice.positions[i].x] = action_choice.paths[i].prob
                     elif i < len(action_choice.rolls):
                         # Convert to chance of succeeding
-                        chance = 1.0
-                        for roll in action_choice.rolls[i]:
-                            chance = chance * ((1+(6-roll)) / 6)
+                        if action_choice.action_type == ActionType.FOUL:
+                            # Use different probability calculation for 2D6 rolls
+                            chance = self.accumulated_prob_2d_roll(action_choice.rolls[0][0])
```

Comment:
First zero in the index should be "i", I think.

Reply:
Yes. chance = chance * ((1+6-roll) / 6) looks a bit cleaner.